core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3151,9 +3152,31 @@ private[spark] object Utils
   }
 }
 
-private[util] object CallerContext extends Logging {
```

Comment:
I needed to relax visibility on things here to facilitate unit testing with caller context enabled. LMK if a different approach is preferred (reflection?).

Reply:
This change looks okay for me.